client.go

```diff
@@ -2006,16 +2005,37 @@ func (c *HostClient) getClientName() []byte {
 	return clientName
 }
 
-func addMissingPort(addr string, isTLS bool) string {
-	n := strings.Index(addr, ":")
-	if n >= 0 {
+func AddMissingPort(addr string, isTLS bool) string {
```

Comment:
```
// AddMissingPort adds a port to a host if it is missing.
// A literal IPv6 address in hostport must be enclosed in square
// brackets, as in "[::1]:80", "[::1%lo0]:80".
func AddMissingPort(addr string, isTLS bool) string {
	if strings.IndexByte(addr, ':') >= 0 {
		endOfV6 := strings.IndexByte(addr, ']')
		// we do not care about the validity of the address, just check if it has more bytes after ']'
		if endOfV6 < len(addr)-1 {
			return addr
		}
	}
	if !isTLS {
		return addr + ":80"
	}
	return addr + ":443"
}

```
This is a much cleaner implementation and remember to add comments and Unit tests to AddMissingPort.

Reply:
Thank you, your version is cleaner. However it won't work for IPv6 without brackets i.e. `::1` but we don't know if the plain IPv6 addresses are already in use.
My version may be not so clean or efficient but more safe.

Reply:
> Thank you, your version is cleaner. However it won't work for IPv6 without brackets i.e. `::1` but we don't know if the plain IPv6 addresses are already in use. My version may be not so clean or efficient but more safe.

However, in the case of web development scenarios, according to RFC https://www.ietf.org/rfc/rfc2732.txt, [] is definitely present,

Reply:
Yes, totally agree. Seems like we safe here. Even if someone creates manually the HostClient with the ipv6 without brackets it will work fine as previously. I'll change the pr later to your version.